.github/workflows/build.yml

```diff
@@ -0,0 +1,733 @@
+        working-directory: ${{ steps.build_paths.outputs.DOWNLOADS }}
+        run: |
+          folder_name="vcpkg_${{ matrix.image.name }}-${{ matrix.image.tag }}_llvm-${{ matrix.llvm }}_amd64"
+          archive_name="${folder_name}.tar.xz"
+
+          url="https://github.com/trailofbits/cxx-common/releases/download/${{ matrix.cxxcommon_version}}/${archive_name}"
+          local_path="${{ steps.build_paths.outputs.DOWNLOADS }}/${archive_name}"
+
+          if [[ ! -f "${local_path}" ]]; then
+            echo "Downloading: ${url}"
+            curl "${url}" -L -o "${local_path}"
+
+            ls -t ${{ steps.build_paths.outputs.DOWNLOADS }}/vcpkg_* | tail -n +2 | while read archive_file ; do
+              rm "${archive_file}"
+            done
+          fi
+
+          tar xf "${local_path}" \
+              -C "${{ steps.build_paths.outputs.INSTALL }}"
+
+          destination_path="${{ steps.build_paths.outputs.INSTALL }}/vcpkg"
+
+          mv "${{ steps.build_paths.outputs.INSTALL }}/${folder_name}" \
+             "${destination_path}"
+
+          echo "${destination_path}/installed/x64-linux/tools/llvm-11" >> $GITHUB_PATH
+
+          echo ::set-output name=PATH::${destination_path}
+
+      - name: Select the build job count
+        shell: bash
+        id: build_job_count
+        run: |
+          echo ::set-output name=VALUE::$(($(nproc) + 1))
+
+      - name: Configure remill
+        working-directory: ${{ steps.build_paths.outputs.BUILD }}
+
+        env:
+          CCACHE_DIR: ${{ steps.build_paths.outputs.CCACHE }}
+
+        run: |
+          mkdir remill_build && cd remill_build
+
+          cmake \
+              -DCMAKE_VERBOSE_MAKEFILE=true \
+              -DVCPKG_ROOT="${{ steps.cxxcommon_installer.outputs.PATH }}" \
+              -G Ninja \
+              ${{ steps.build_paths.outputs.SOURCE }}/remill
+
+      - name: Build remill
+        working-directory: ${{ steps.build_paths.outputs.BUILD }}
+
+        env:
+          CCACHE_DIR: ${{ steps.build_paths.outputs.CCACHE }}
+
+        run: |
+          cmake --build remill_build \
+                -j ${{ steps.build_job_count.VALUE }}
+
+      - name: Install remill
+        id: remill_installer
+        working-directory: ${{ steps.build_paths.outputs.BUILD }}
+
+        env:
+          CCACHE_DIR: ${{ steps.build_paths.outputs.CCACHE }}
+
+        run: |
+          export DESTDIR=${{ steps.build_paths.outputs.INSTALL }}/remill
+
+          cmake --build remill_build \
+                --target install
+
+          echo ::set-output name=PATH::${DESTDIR}
+
+      - name: Configure anvill
+        working-directory: ${{ steps.build_paths.outputs.BUILD }}
+
+        env:
+          CCACHE_DIR: ${{ steps.build_paths.outputs.CCACHE }}
+
+        run: |
+          mkdir anvill_build && cd anvill_build
+
+          cmake \
+              -DCMAKE_VERBOSE_MAKEFILE=true \
+              -DVCPKG_ROOT="${{ steps.cxxcommon_installer.outputs.PATH }}" \
```

Comment:
A more standard way would be to use the CMake toolchain specification, which is also supported by vcpkg; dependencies do not normally try to force their own toolchain when linking against them.

I think we should look into this, and also fix support for the system toolchain (we are currently not able to use libFuzzer or the sanitizers with this setup)

Reply:
I see where you're coming from, and I agree for the most part. I implemented it this way since we were trying to minimize the amount of CMake options, since using vcpkg should actually require using `-DCMAKE_TOOLCHAIN_FILE==[path to vcpkg]/scripts/buildsystems/vcpkg.cmake` and also `-DVCPKG_TARGET_TRIPLET=x64-{osx,linux}-rel` being passed to every tool. Specifying these is more correct, but could lead to confusion or typos when specifying for all of the different tools, so I tried to wrap these definitions in the `cmake/vcpkg_helper.cmake` script.

vcpkg also has the `-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=` option to load additional toolchain files, but I haven't tested that feature too extensively. Would you be able to provide an example workflow that another project uses for libFuzzer and the sanitizers? I can do some experiments with what it would look like using vcpkg